packages/hooks/src/useActiveSymbols.ts

```diff
@@ -0,0 +1,27 @@
+import { useFetch } from '@deriv/api';
+import { useMemo } from 'react';
+
+/** A custom hook to get the list of active symbols. */
+const useActiveSymbols = () => {
+    const { data, ...rest } = useFetch('active_symbols', {
+        payload: { active_symbols: 'brief' },
+        options: { refetchOnWindowFocus: false },
+    });
+    const active_symbols = useMemo(
```

Comment:
```suggestion
    const active_symbols = useMemo(() => {
    const active_symbols = data?.active_symbols
    
    if(!active_symbols) return undefined
    
    return active_symbols.map(....
```

Reply:
@niloofar-deriv since we have a separate hook to fetch the data, I think this is fine if we guarantee that this will return an array, if there's no data then empty array should be good. The reason is, person who is using the hook need to add extra checks inside the component where he is trying to use the `active_symbols` data

Reply:
@amir-deriv Having the extra check is not a bad thing, It makes the code safer actually. Also returning default values in the data layer is hidden error handling, Like we don't have any data but return an empty array which is data and might mean something.

As an example, We have places in BE where data can be `0 | 1 | undefined`, We transform this to a boolean but if we ignore the `undefined` part we will be doing silent error handling. And what happens is that the `undefined` actually means something as the third possible value in BE response, And now we always ignore it and this will be missed and no one can debug the code to find it since it's in the data layer 😅 We are doing lots of this in our stores, I can show you examples of what is happening if you are interested 👀 

I think what @niloofar-deriv suggested is a better approach 🙇🏻

Reply:
> As an example, We have places in BE where data can be `0 | 1 | undefined`
is this on endpoints which is expecting data as an array and got these values ? if that's the case then it's fine to return `undefined`

but for the error handling part, we are already handling error in our `useFetch` hook like follows

```js
    const response = await WS.send({ [name]: 1, ...(payload || {}) });

    if (response.error) {
        throw response.error;
    }

```

so the error coming from BE will change the state in our hook, my concern is, basically when we want to fetch data from BE in a traditional way, we go like this:

1. we have some initial state, in this case let's say `active_symbols = []`
2. we fetch the data, if the request is successful we change the state of `acitve_symbols` to the values coming from the response

3. if there's any error, we handle it accordingly, (**note that we are not touching the value of `active_symbols` it's still an empty array and has nothing to do with how we handle the error**)

I understand that if in some places we get `0 | 1 | undefined` we shouldn't ignore undefined, and if it's the same case with `active_symbols` that we may get `0 | 1 | undefined` instead of some error. then we should leave not add the default value. but if this is not the case then default value is fine too


cc: @farzin-deriv @niloofar-deriv

Reply:
@amir-deriv @niloofar-deriv Let's jump on a call and discuss it if you guys are interested 🙇🏻

Reply:
Yes sure @farzin-deriv